src/winsvcaux.c

```diff
@@ -51,9 +51,20 @@ static int lua_GetErrorString(lua_State *L)
     (LPTSTR)&lpMsgBuf,
     0, NULL);
 
-  lua_pushstring(L, lpMsgBuf);
-  LocalFree(lpMsgBuf);
-  return 1;
+  if (len) {
+      if (len > 2) {
+          // strip \r\n
+          lpMsgBuf[len - 2] = '\0';
+      }
+      lua_pushstring(L, lpMsgBuf);
+      lua_pushnil(L);
+      LocalFree(lpMsgBuf);
+  }
+  else {
+      lua_pushstring(L, "");
```

Comment:
i guess pushing nil is ok, i wanted to give that the most chance of succeeding when something was expecting a string return